vllm/engine/arg_utils.py

```diff
@@ -550,6 +558,27 @@ def from_cli_args(cls, args: argparse.Namespace):
         return engine_args
 
     def create_engine_config(self, ) -> EngineConfig:
+
+        # bitsandbytes quantization is unique in that it needs a specific model loader
+        # so we need to make sure the quant method and the load format are consistent
+        if self.quantization == "bitsandbytes" or self.qlora_adapter_name_or_path is not None:
+            if self.load_format == "auto":
+                self.load_format = "bitsandbytes"
+            else:
+                if self.load_format != "bitsandbytes":
+                    raise ValueError(
+                        f"BitsAndBytes quantization cannot work with {self.load_format} load format"
+                    )
+
+        if self.load_format == "bitsandbytes" or self.qlora_adapter_name_or_path is not None:
+            if self.quantization == None:
+                self.quantization = "bitsandbytes"
+            else:
+                if self.quantization != "bitsandbytes":
+                    raise ValueError(
+                        f"BitsAndBytes load format cannot work with {self.quantization} "
+                    )
```

Comment:
let's just throw an exception if load format is bitsandbytes but quantization is not (and tell the user to fix it themselves)

Reply:
fixed.